### PR TITLE
Improve waitForPDBCollectionToBeDeleted in e2e test

### PR DESCRIPTION
**Pull request by k-toyoda-pi**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
I believe that `waitForPDBCollectionToBeDeleted` intends to wait for PDB collection being deleted.
However, it does not wait and finishes with error immediately if items is not 0.
This PR changes that function to wait for deletion.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

File: test/e2e/apps/disruption.go [hunk 1]
```diff
@@ -431,15 +431,17 @@ func deletePDBCollection(cs kubernetes.Interface, ns string) {
 
 func waitForPDBCollectionToBeDeleted(cs kubernetes.Interface, ns string) {
 	ginkgo.By("Waiting for the PDB collection to be deleted")
-	wait.PollImmediate(framework.Poll, schedulingTimeout, func() (bool, error) {
+	err := wait.PollImmediate(framework.Poll, schedulingTimeout, func() (bool, error) {
 		pdbList, err := cs.PolicyV1beta1().PodDisruptionBudgets(ns).List(context.TODO(), metav1.ListOptions{})
 		if err != nil {
 			return false, err
 		}
-		framework.ExpectNoError(err, "Listing PDB set in namespace %s", ns)
```
Comment (oomichi) [inline test/e2e/apps/disruption.go]: Nice catch.

**Review by oomichi — COMMENTED** [review]
/lgtm

File: test/e2e/apps/disruption.go (hunk 1 above)
Comment (k-toyoda-pi) [reply test/e2e/apps/disruption.go]: Thank you for reviewing.

**Comment by k-toyoda-pi** [comment]
@nan-yu @mortent Hello, I'm updating e2e sub func for PDB, could you make sure if this matches your intention?

**Review by oomichi — COMMENTED** [review]
/approve

**Pull request merged**